src/inpage/inpage.ts

```diff
@@ -116,7 +116,7 @@ const provider: StacksProvider = {
     const event = new CustomEvent<AuthenticationRequestEventDetails>(
       DomEventName.authenticationRequest,
       {
-        detail: { authenticationRequest },
+        detail: { authenticationRequest, flowInitiatorUrl: document.URL },
```

Comment:
This is problematic in that we're trusting the `inpage.ts` file. This runs in the actual context of the webpage, and could therefore be changed set to anything by the page's author (presenting a security problem if this were later trusted in the app).

Instead, we should use the URLs that are included in the `sender` parameter where the events are listened to in the background script. In addition to `origin` you can see the fullpage url.